kakao/KakaoLoginAndroid/app/src/main/java/com/google/firebase/auth/kakao/MainActivity.java

```diff
@@ -0,0 +1,217 @@
+        FirebaseUser currentUser = FirebaseAuth.getInstance().getCurrentUser();
+        if (currentUser != null) {
+            binding.setCurrentUser(currentUser);
+            if (currentUser.getPhotoUrl() != null) {
+                imageView.setImageUrl(currentUser.getPhotoUrl().toString(), new ImageLoader(Volley.newRequestQueue(getApplicationContext()), new ImageLoader.ImageCache() {
+                    private final LruCache<String, Bitmap> cache = new LruCache<String, Bitmap>(20);
+                    @Override
+                    public Bitmap getBitmap(String url) {
+                        return cache.get(url);
+                    }
+
+                    @Override
+                    public void putBitmap(String url, Bitmap bitmap) {
+                        cache.put(url, bitmap);
+                    }
+                }));
+            }
+            loginButton.setVisibility(View.INVISIBLE);
+            loggedInView.setVisibility(View.VISIBLE);
+            logoutButton.setVisibility(View.VISIBLE);
+        } else {
+            loginButton.setVisibility(View.VISIBLE);
+            loggedInView.setVisibility(View.INVISIBLE);
+            logoutButton.setVisibility(View.INVISIBLE);
+        }
+    }
+
+    /**
+     * OnActivityResult() should be overridden for Kakao Login because Kakao Login uses startActivityForResult().
+     */
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        Session.getCurrentSession().handleActivityResult(requestCode, resultCode, data);
+    }
+
+    /**
+     *
+     * @param kakaoAccessToken Access token retrieved after successful Kakao Login
+     * @return Task object that will call validation server and retrieve firebase token
+     */
+    private Task<String> getFirebaseJwt(final String kakaoAccessToken) {
+        final TaskCompletionSource<String> source = new TaskCompletionSource<>();
+
+        RequestQueue queue = Volley.newRequestQueue(this);
+        String url = getResources().getString(R.string.validation_server_domain) + "/verifyToken";
+        HashMap<String, String> validationObject = new HashMap<>();
+        validationObject.put("token", kakaoAccessToken);
+
+        JsonObjectRequest request = new JsonObjectRequest(Request.Method.POST, url, new JSONObject(validationObject), new Response.Listener<JSONObject>() {
+            @Override
+            public void onResponse(JSONObject response) {
+                try {
+                    String firebaseToken = response.getString("firebase_token");
+                    source.setResult(firebaseToken);
+                } catch (Exception e) {
+                    source.setException(e);
+                }
+
+            }
+        }, new Response.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError error) {
+                Logger.e(error.toString());
```

Comment:
nit: use the normal Android logcat:

`Log.e(TAG, "Error getting firebase token from server", e)`